### PR TITLE
PP-8945 Store fees and send event for failed payments

### PR DESCRIPTION
**Pull request by stephencdaly**
When processing Stripe failed payments, after transferring the applicable amount for Stripe fees from the Stripe connect account, store entries in the `fees` table for each of the fee types that were applied. Send a FEE_INCURRED event to ledger containing the fee breakdown.

Always expect that the charge exists in the connector database, and throw an exception if it has been expunged. We will separately add a check that will stop Stripe charges from being expunged if they have not had the applicable fees taken.

Stop using StripePaymentProvider and call StripeFailedPaymentFeeCollectionHandler directly from
TaskQueueMessageHandler to simplify the code. To allow for this, added a new GatewayOperation enum entry so a separate Stripe client can be created in StripeFailedPaymentFeeCollectionHandler.

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/gateway/stripe/handler/StripeFailedPaymentFeeCollectionHandler.java [hunk 1]
```diff
@@ -1,53 +1,93 @@
...
     private final StripeGatewayConfig stripeGatewayConfig;
     private final JsonObjectMapper jsonObjectMapper;
+    private final Clock clock;
+    private final EventService eventService;
+    private final ChargeDao chargeDao;
 
+    @Inject
     public StripeFailedPaymentFeeCollectionHandler(
-            GatewayClient gatewayClient,
+            GatewayClientFactory gatewayClientFactory,
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/connector/gateway/stripe/handler/StripeFailedPaymentFeeCollectionHandler.java]: Any reason db operations and event emitting is done here ? Looks like StripeFailedPaymentFeeCollectionHandler is responsible for all operations (stripe, db, and emitting events) and doesn't feel like this should be under a gateway specific package gateway/stripe/handler/... . May be a service layer that can use StripePaymentProvider for gateway operation and does rest of the operations ?

Comment (stephencdaly) [reply src/main/java/uk/gov/pay/connector/gateway/stripe/handler/StripeFailedPaymentFeeCollectionHandler.java]: updated to separate out the gateway operations

File: src/main/java/uk/gov/pay/connector/gateway/stripe/request/StripeGetPaymentIntentRequest.java [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 import uk.gov.pay.connector.app.StripeGatewayConfig;
 import uk.gov.pay.connector.charge.model.domain.Charge;
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/connector/gateway/stripe/request/StripeGetPaymentIntentRequest.java:4]: Few unused imports

**Review by kbottla — APPROVED** [review]
LGTM!

File: src/main/java/uk/gov/pay/connector/gateway/stripe/request/StripeGetPaymentIntentRequest.java (hunk 2 above)
Comment (stephencdaly) [reply src/main/java/uk/gov/pay/connector/gateway/stripe/request/StripeGetPaymentIntentRequest.java:4]: Thanks, will remove in next PR

**Pull request merged**